src/main/java/net/fabricmc/fabric/mixin/client/render/MixinFluidRenderer.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2016, 2017, 2018 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.client.render;
+
+import net.fabricmc.fabric.api.client.render.FluidRenderHandler;
+import net.fabricmc.fabric.api.client.render.FluidRenderHandlerRegistry;
+import net.fabricmc.fabric.impl.accessors.FluidRendererHookContainer;
+import net.fabricmc.fabric.impl.client.render.FluidRenderHandlerRegistryImpl;
+import net.minecraft.client.render.BufferBuilder;
+import net.minecraft.client.render.block.FluidRenderer;
+import net.minecraft.client.texture.Sprite;
+import net.minecraft.fluid.FluidState;
+import net.minecraft.util.ActionResult;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.ExtendedBlockView;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.ModifyVariable;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+@Mixin(FluidRenderer.class)
+public class MixinFluidRenderer {
+    @Shadow
+    private Sprite[] lavaSprites;
+    @Shadow
+    private Sprite[] waterSprites;
+
+    private final ThreadLocal<FluidRendererHookContainer> fabric_renderHandler = ThreadLocal.withInitial(FluidRendererHookContainer::new);
+
+    @Inject(at = @At("RETURN"), method = "onResourceReload")
+    public void onResourceReloadReturn(CallbackInfo info) {
+        FluidRenderHandlerRegistryImpl.INSTANCE.onFluidRendererReload(waterSprites, lavaSprites);
+    }
+
+    @Inject(at = @At("HEAD"), method = "tesselate", cancellable = true)
+    public void tesselate(ExtendedBlockView view, BlockPos pos, BufferBuilder bufferBuilder, FluidState state, CallbackInfoReturnable<Boolean> info) {
+        FluidRendererHookContainer ctr = fabric_renderHandler.get();
```

Comment:
This would be a hack, but if you want to avoid the additional ThreadLocal lookups you could make your helper wrap/implement ExtendedBlockView (or FluidState if that's easier).  That would mean everything you need is in an LVT instance and so you only need to cast it vs do a ThreadLocal.get().